### PR TITLE
Project restructure

### PR DESCRIPTION
**Pull request by clacroix12**
This is a gigantic PR with the following goal: consolidate existing modules and restructure the project to be more in line with other python projects. More details can be found in #207 . Tox execution was successful as well as a deployment to AWS (will post logs shortly).

### CODE REVIEW

**Comment by clacroix12** [comment]
@zmc is there a functional reason it's an issue or is it just better to keep that in the package since we depend on it in `ocs_ci`? We can totally move that to a new location, but as far as functionality goes I didn't see any issues with it during unit tests or deployment.

**Comment by clacroix12** [comment]
Apparently we are not generating a log file for the deployment anymore since it is a fixture. The only log file I have is the minimal test that essentially tests cluster connectivity. I will open a bug for this issue.
```
2019-06-26 11:04:33,753 - INFO - ocs_ci.ocs.openshift_ops.set_kubeconfig.139 - Testing access to cluster with /Users/clacroix/clusters/rework-2/auth/kubeconfig
2019-06-26 11:04:33,754 - INFO - ocs_ci.utility.utils.run_cmd.361 - Executing command: oc cluster-info
2019-06-26 11:04:34,255 - DEBUG - ocs_ci.utility.utils.run_cmd.371 - CMD output: Kubernetes master is running at https://api.clacroix.qe.rh-ocs.com:6443

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.

2019-06-26 11:04:34,255 - INFO - ocs_ci.ocs.openshift_ops.set_kubeconfig.151 - Access to cluster is OK!
```

**Comment by zmc** [comment]
@clacroix12 yeah, it's really just a packaging issue:
```
$ virtualenv -p python3.7 /tmp/venv && source /tmp/venv/bin/activate && pip install git+https://github.com/clacroix12/ocs-ci.git@project-restructure && python -c 'from ocs_ci.framework import config; print(config.to_dict())'
[...]
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/venv/lib/python3.7/site-packages/conf/ocsci/default_config.yaml'
```

**Comment by clacroix12** [comment]
@zmc I moved it to a new `conf` dir under the `framework` module. Not sure if this is the _best_ place for it but it does address the packaging concerns. Since the config discussion is still ongoing I figured this would be a suitable home for now. We can always relocate/remove/do whatever to it in a subsequent PR.

**Comment by clacroix12** [comment]
Rebased to address merge conflicts.

File: ocs_ci/run_ocsci.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import sys
 
-import ocsci.main
+from .framework import main as runner
```
Comment (ebenahar) [inline ocs_ci/run_ocsci.py]: I think it would be better to keep absolute importing everywhere

**Comment by mbukatov** [comment]
@clacroix12 thank's for driving this. I will provide my review you asked me for tomorrow morning CEST.

I have some python packaging and setuptools experience, and plan to suggest some changes related to that if needed.

File: ocs_ci/run_ocsci.py (hunk 1 above)
Comment (clacroix12) [reply ocs_ci/run_ocsci.py]: Done

**Comment by clacroix12** [comment]
Rebased (no changes) to fix merge conflicts due to #304 .

**Comment by zmc** [comment]
@clacroix12 Your updates re: the config file get us most of the way, but we should also add this line to `MANIFEST.in`:
```
recursive-include ocs_ci/framework/conf *.yaml
```

**Comment by clacroix12** [comment]
> @clacroix12 Your updates re: the config file get us most of the way, but we should also add this line to `MANIFEST.in`:
> 
> ```
> recursive-include ocs_ci/framework/conf *.yaml
> ```

@zmc We use `graft` for the `conf` directory currently. Is pattern matching with `recursive-include` a better option? Also do we need to do something similar for our `templates`?

**Comment by zmc** [comment]
@clacroix12 The `graft` line is picking up the top-level `conf` directory, but not `ocs_ci/framework/conf/default_config.yaml` - preventing a packaged version of the project from working. If we want to ship `conf/` we should keep the `graft`, sure.

It does look like several more things are missing, like `ocs_ci/templates/`, `docs/`, `pytest*.ini`, and `tox.ini` (not necessarily an exhaustive list).

IMO it's ok to let a few of these things slide for now, but we should at least fix what we know will actually break packaged versions of the project. In the future we could choose to catch issues by `pip install`ing (even from git) in Jenkins, as opposed to doing a `git clone`.

**Comment by mbukatov** [comment]
@clacroix12 @zmc 
> IMO it's ok to let a few of these things slide for now, but we should at least fix what we know will actually break packaged versions of the project. In the future we could choose to catch issues by `pip install`ing (even from git) in Jenkins, as opposed to doing a `git clone`.

This could be easily addressed by adding `graft ocs_ci` into `MANIFEST.in` (doing this makes sense because there are many static non python files in this directory tree).

Then it just adds anything there into sdist tarball:

```
$ python3 setup.py sdist
$ tar tf dist/ocs-ci-0.1.tar.gz | grep conf
ocs-ci-0.1/conf/
ocs-ci-0.1/conf/README.md
ocs-ci-0.1/conf/examples/
ocs-ci-0.1/conf/examples/ocs_basic_install.yml
ocs-ci-0.1/conf/ocs_basic_install.yml
ocs-ci-0.1/conf/ocsci/
ocs-ci-0.1/conf/ocsci/README.md
ocs-ci-0.1/conf/ocsci/downstream_config.yaml
ocs-ci-0.1/ocs_ci/framework/conf/
ocs-ci-0.1/ocs_ci/framework/conf/default_config.yaml
ocs-ci-0.1/ocs_ci/framework/tests/test_config.py
ocs-ci-0.1/ocs_ci/templates/install-config.yaml.j2
ocs-ci-0.1/tests/conftest.py
```

**Comment by clacroix12** [comment]
@RazTamir Rebased and re-ran deployment + tier1 tests. As far as I can tell the test status is comparable to what @petr-balogh showed in his "state of tier1 tests" email. The main point here is that we can execute deployment, tier1, and teardown with comparable results to what we are seeing on master currently. Since we are going to handle cleaning up the status of our tests externally I think this PR is good to go now. @mbukatov @zmc I have used graft to include `ocs_ci` in our manifest as well as removed the PYTHONPATH from our tox.ini. 

```
Matrix Test Report
===================
                                                                                                                         report.xml
                                                                                                                         |
tests.test_cluster_utils.TestClusterUtils
- test_add_mds                                                                                                           f  F Total Failure
- test_add_mon                                                                                                           f  F Total Failure
- test_get_admin_key                                                                                                     /  / Passed
- test_get_mon_info                                                                                                      /  / Passed
- test_get_user_key                                                                                                      /  / Passed
tests.test_example_of_pytest_test.TestExampleClass
- test_example_method                                                                                                    s  U Untested
tests.manage.test_add_mds_to_cluster.TestCephFilesystemCreation
- test_cephfilesystem_creation                                                                                           f  F Total Failure
tests.manage.test_create_pvc_random_storage_class.TestCreatePVCRandomStorageClass
- test_create_pvc_with_random_sc                                                                                         f  F Total Failure
tests.manage.test_create_storage_class_pvc.TestOSCBasics
- test_basics_cephfs                                                                                                     f  F Total Failure
- test_basics_rbd                                                                                                        f  F Total Failure
tests.manage.test_create_storageclass_with_same_name.TestCaseOCS322
- test_create_storageclass_with_same_name                                                                                /  / Passed
tests.manage.test_pv_creation.TestPvCreation
- test_pv_creation                                                                                                       /  / Passed
tests.manage.test_pvc_concurrent_deletion_creation.TestMultiplePvcConcurrentDeletionCreation
- test_multiple_pvc_concurrent_creation_deletion                                                                         f  F Total Failure
tests.manage.test_pvc_delete_verify_size_is_returned_to_backendpool.TestPVCDeleteAndVerifySizeIsReturnedToBackendPool
- test_pvc_delete_and_verify_size_is_returned_to_backend_pool                                                            f  F Total Failure
tests.manage.test_pvc_deletion_during_io.TestDeletePVCWhileRunningIO
- test_run_io_and_delete_pvc                                                                                             f  F Total Failure
tests.manage.test_rbd_csi_default_sc.TestBasicPVCOperations
- test_ocs_347                                                                                                           f  F Total Failure

-------------------------------------------------------------------------------
Test Results:
  Failed       :     13
  Passed       :      5
  Skipped      :      1
```

**Pull request merged**